packages/moonstone/Scrollable/ScrollButtons.js

```diff
@@ -236,36 +236,50 @@ class ScrollButtons extends Component {
 	onSpotlight = (ev) => {
 		const
 			{focusableScrollButtons, rtl, vertical} = this.props,
-			{target} = ev;
+			{target} = ev,
+			direction = getDirection(ev.keyCode),
+			isDown = direction === 'down',
+			isLeftMovement = direction === (rtl ? 'right' : 'left'),
+			isRightMovement = direction === (rtl ? 'left' : 'right'),
+			isUp = direction === 'up',
+			isNextButton = target === this.nextButtonRef.current,
+			isPrevButton = target === this.prevButtonRef.current;
 
 		// We don't need to navigate manually if `focusableScrollButtons` is `false`
 		if (focusableScrollButtons) {
 			const
-				direction = getDirection(ev.keyCode),
-				fromNextToPrev = (vertical && direction === 'up') || (!vertical && direction === (rtl ? 'right' : 'left')),
-				fromPrevToNext = (vertical && direction === 'down') || (!vertical && direction === (rtl ? 'left' : 'right'));
+				fromNextToPrev = vertical ? isUp : isLeftMovement,
+				fromPrevToNext = vertical ? isDown : isRightMovement;
 
 			// manually focus the opposite scroll button when 5way pressed
-			if ((fromNextToPrev && target === this.nextButtonRef.current) ||
-				(fromPrevToNext && target === this.prevButtonRef.current)) {
+			if (isNextButton && fromNextToPrev || isPrevButton && fromPrevToNext) {
 				this.focusOnOppositeScrollButton(ev, direction);
 			}
 		} else {
-			// If it is vertical `Scrollable`, move focus to the left for ltr or to the right for rtl
-			// If is is horizontal `Scrollable`, move focus to the up
-			const direction = !vertical && 'up' || rtl && 'right' || 'left';
-
-			if (Spotlight.getPointerMode()) {
-				// When changing from "pointer" mode to "5way key" mode,
-				// a pointer is hidden and a last focused item get focused after 30ms.
-				// To make sure the content in `VirtualList` or `Scroller` to be focused after that, we used 50ms.
-				setTimeout(() => {
-					if (Spotlight.getCurrent() === target) {
-						Spotlight.move(direction);
+			const
+				// If it is vertical `Scrollable`, move focus to the left for ltr or to the right for rtl
+				// If is is horizontal `Scrollable`, move focus to the up
+				directionToContent = !vertical && 'up' || rtl && 'right' || 'left',
+				isDirectionToLeave =
+					vertical && (isRightMovement || isPrevButton && isUp || isNextButton && isDown) ||
+					!vertical && (isDown || isPrevButton && isLeftMovement || isNextButton && isRightMovement);
+
+			if (isDirectionToLeave) {
+				preventDefault(ev);
+				if (!Spotlight.move(direction)) {
+					if (Spotlight.getPointerMode()) {
```

Comment:
When I checked, `Spotlight.getPointerMode()` is always `false`. When could `Spotlight.getPointerMode()` value be `true`?

Reply:
@ybsung Good catch! I should have checked the existing code first. In short, this routine will run always in 5-way mode. I will remove this routine.

Here is a short history. When this routine is born (#1776), we need to wait 5-way mode and put `setTimeout` here. Then, after another patch for navigation on disabled items (#1778), pointer mode is turned off within a keydown handler in VirtualListBase. Because of not only bubbling order but also the order between a DOM event handler (VirtualListBase) and React synthetic event handler (Spottable->ScrollButtons), ScrollButton's handler runs always on 5-way mode.